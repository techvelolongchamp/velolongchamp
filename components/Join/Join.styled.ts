import styled from 'styled-components'

export const ButtonContainer = styled.div`
  text-align: center;
  margin: ${({ theme }) => theme.spacing.l};
`

export const ButtonLink = styled.a`
  background: ${({ theme }) => theme.colors.brandSecondary};
  border-radius: 5px;
  color: ${({ theme }) => theme.colors.white};
  cursor: pointer;
  font-size: ${({ theme }) => theme.fonts.large};
  margin: ${({ theme }) => theme.spacing.l} ${({ theme }) => theme.spacing.s};
  padding: ${({ theme }) => theme.spacing.s};
`

export const JoinContainer = styled.article`
  padding-bottom: ${({ theme }) => theme.spacing.l};

  p {
    color: ${({ theme }) => theme.colors.defaultColor};
    font-size: ${({ theme }) => theme.fonts.medium};
    line-height: ${({ theme }) => theme.spacing.l};
    padding: ${({ theme }) => theme.spacing.s} 0;
    text-align: justify;
  }

  ol {
    counter-reset: styled-counter;
    display: flex;
    flex-wrap: wrap;
    list-style: none;
    margin: 0;
    padding: 0;
  }

  ol li {
    align-items: center;
    counter-increment: styled-counter;
    display: flex;
    margin: ${({ theme }) => theme.spacing.s} 0;
    width: 100%;
    p {
      color: ${({ theme }) => theme.colors.brandSecondary};
      font-size: ${({ theme }) => theme.fonts.large};
      font-weight: 500;
      line-height: 20px;
      padding-right: ${({ theme }) => theme.spacing.s};
    }
  }

  ol li::before {
    align-items: center;
    color: ${({ theme }) => theme.colors.brandSecondary};
    content: '0' counter(styled-counter);
    display: flex;
    font-family: 'Abril Fatface', serif;
    font-size: ${({ theme }) => theme.fonts.xlarge};
    font-weight: bold;
    line-height: 1;
    margin-right: ${({ theme }) => theme.spacing.s};
  }
`

export const Faq = styled.article`
  padding-bottom: ${({ theme }) => theme.spacing.xl};
  h3 {
    color: ${({ theme }) => theme.colors.brandTertiary};
    font-size: ${({ theme }) => theme.fonts.large};
    font-weight: 600;
    margin: ${({ theme }) => theme.spacing.l} 0;
  }
`

export const Question = styled.p`
  color: ${({ theme }) => theme.colors.defaultColor};
  font-size: ${({ theme }) => theme.fonts.medium};
  font-style: italic;
  font-weight: 500;
  line-height: ${({ theme }) => theme.spacing.l};
  margin-top: ${({ theme }) => theme.spacing.s};
  padding: ${({ theme }) => theme.spacing.s} 0;
`

export const Answer = styled.p`
  color: ${({ theme }) => theme.colors.defaultColor};
  font-size: ${({ theme }) => theme.fonts.medium};
  font-weight: 200;
  line-height: ${({ theme }) => theme.spacing.l};
  padding: ${({ theme }) => theme.spacing.s} 0;
`
