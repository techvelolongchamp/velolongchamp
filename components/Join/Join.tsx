import React from 'react'
import Link from 'next/link'
import { useIntl } from 'react-intl'

import {
  ButtonContainer,
  ButtonLink,
  JoinContainer,
  Faq,
  Question,
  Answer,
} from './Join.styled'

const Join = () => {
  const { formatMessage } = useIntl()
  return (
    <>
      <JoinContainer>
        <p>{formatMessage({ id: 'joinUs.joinMessage' })}</p>
        <ButtonContainer>
          <Link href="/join" passHref>
            <ButtonLink>{formatMessage({ id: 'joinUs.clickJoin' })}</ButtonLink>
          </Link>
        </ButtonContainer>

        <p>{formatMessage({ id: 'joinUs.reasonToJoin.title' })}</p>
        <ol>
          {formatMessage(
            { id: 'joinUs.reasonToJoin.list' },
            {
              li: (chunk) => (
                <li>
                  <p>{chunk}</p>
                </li>
              ),
              strong: (chunk) => <strong>{chunk}</strong>,
            }
          )}
        </ol>
      </JoinContainer>
      <Faq>
        <h3>{formatMessage({ id: 'joinUs.faq.title' })}</h3>
        {formatMessage(
          { id: 'joinUs.faq.questions' },
          {
            q: (chunk) => <Question>{chunk}</Question>,
            a: (chunk) => <Answer>{chunk}</Answer>,
          }
        )}
      </Faq>
    </>
  )
}

export default Join
