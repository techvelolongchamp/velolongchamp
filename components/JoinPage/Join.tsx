import React from 'react'
import { useIntl } from 'react-intl'

import { SubTitle, FrameContainer, Message } from './Join.styled'

const Join = () => {
  const { formatMessage } = useIntl()
  return (
    <>
      <SubTitle>{formatMessage({ id: 'join.subtitle' })}</SubTitle>
      <FrameContainer>
        <iframe
          title={formatMessage({ id: 'join.title' })}
          id="haWidget"
          data-hj-allow-iframe=""
          scrolling="auto"
          src="https://www.helloasso.com/associations/velo-longchamp/adhesions/adhesion-association-velo-longchamp/widget"
          style={{ width: '100%', height: '750px', border: 'none' }}
        ></iframe>
      </FrameContainer>
      <Message>
        {formatMessage(
          { id: 'join.problem' },
          {
            a: (chunk: React.ReactNode) => (
              <a
                href="https://www.helloasso.com/associations/velo-longchamp/adhesions/adhesion-association-velo-longchamp"
                target="_blank"
                rel="noopener noreferrer"
                title="HelloAsso"
              >
                {chunk}
              </a>
            ),
          }
        )}
      </Message>
    </>
  )
}

export default Join
