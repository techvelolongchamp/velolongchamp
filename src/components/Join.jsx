import React from 'react'
import styled from 'styled-components'
import { OutboundLink } from 'gatsby-plugin-google-analytics'

const ButtonContainer = styled.div`
  text-align: center;
  margin: ${({ theme }) => theme.spacing.l};
`

const ButtonLink = styled(OutboundLink)`
  margin: ${({ theme }) => theme.spacing.l} ${({ theme }) => theme.spacing.s};
  padding: ${({ theme }) => theme.spacing.s};
  font-size: ${({ theme }) => theme.fonts.large};
  color: ${({ theme }) => theme.colors.white};
  background: ${({ theme }) => theme.colors.brandSecondary};
  border-radius: 5px;
`

const JoinContainer = styled.article`
  padding-bottom: ${({ theme }) => theme.spacing.l};

  p {
    font-size: ${({ theme }) => theme.fonts.medium};
    line-height: ${({ theme }) => theme.spacing.l};
    color: ${({ theme }) => theme.colors.defaultColor};
    padding: ${({ theme }) => theme.spacing.s} 0;
    text-align: justify;
  }

  ol {
    list-style: none;
    counter-reset: styled-counter;
    display: flex;
    flex-wrap: wrap;
    margin: 0;
    padding: 0;
  }

  ol li {
    counter-increment: styled-counter;
    display: flex;
    align-items: center;
    margin: ${({ theme }) => theme.spacing.s} 0;
    width: 100%;
    h4 {
      line-height: 20px;
      color: ${({ theme }) => theme.colors.brandSecondary};
      font-size: ${({ theme }) => theme.fonts.large};
      font-weight: 500;
      padding-right: ${({ theme }) => theme.spacing.s};
    }
  }

  ol li::before {
    align-items: center;
    color: ${({ theme }) => theme.colors.brandSecondary};
    content: '0' counter(styled-counter);
    display: flex;
    font-family: 'Abril Fatface', serif;
    font-size: ${({ theme }) => theme.fonts.xlarge};
    font-weight: bold;
    line-height: 1;
    margin-right: ${({ theme }) => theme.spacing.s};
  }
`

const Faq = styled.article`
  padding-bottom: ${({ theme }) => theme.spacing.xl};
  h3 {
    font-size: ${({ theme }) => theme.fonts.large};
    margin: ${({ theme }) => theme.spacing.l} 0;
    color: ${({ theme }) => theme.colors.brandTertiary};
    font-weight: 600;
  }
`

const Question = styled.p`
  color: ${({ theme }) => theme.colors.defaultColor};
  font-size: ${({ theme }) => theme.fonts.medium};
  font-style: italic;
  font-weight: 500;
  line-height: ${({ theme }) => theme.spacing.l};
  margin-top: ${({ theme }) => theme.spacing.s};
  padding: ${({ theme }) => theme.spacing.s} 0;
`

const Answer = styled.p`
  font-size: ${({ theme }) => theme.fonts.medium};
  line-height: ${({ theme }) => theme.spacing.l};
  color: ${({ theme }) => theme.colors.defaultColor};
  padding: ${({ theme }) => theme.spacing.s} 0;
  font-weight: 200;
`

const Join = () => (
  <>
    <JoinContainer>
      <p>
        Devenir membre de VELO LONGCHAMP est gratuit et ne prend que quelques
        instants:
      </p>
      <ButtonContainer>
        <ButtonLink
          href="https://www.helloasso.com/associations/velo-longchamp"
          target="_blank"
          rel="noopener noreferrer"
        >
          Rejoindre l'association
        </ButtonLink>
      </ButtonContainer>

      <p>Voici 5 arguments imparables pour vous convaincre d’adhérer :</p>
      <ol>
        <li>
          <h4>
            Vous roulerez plus tôt sur un <b>anneau sécurisé et confortable</b>
             ! Car plus on est nombreux, plus on a de poids et plus vite seront
            faits les travaux d’aménagement nécessaires
          </h4>
        </li>
        <li>
          <h4>
            Votre <b>avis n’est pas moins important</b> que celui d’un autre !
            En devenant membre, vous pouvez exprimer directement vos besoins et
            vos priorités ; vous pouvez lancer une initiative qui vous tient à
            cœur ; vous pouvez voter en Assemblée Générale ; vous êtes tenu au
            courant des avancées du projet
          </h4>
        </li>
        <li>
          <h4>
            Vous pouvez <b>cumuler avec d’autres clubs</b> ! On n’est pas
            exclusif ! Vous pouvez rester membre de votre club de vélo, de
            triathlon ou autre.
          </h4>
        </li>
        <li>
          <h4>
            Vous ne roulerez <b>plus jamais seul</b> !  En nous rejoignant, vous
            allez rencontrer d’autres cyclistes qui pourront, si vous le voulez,
            vous accompagner à Longchamp. Nous allons créer des ateliers
            (initiation tandem, "premiers secours" ou "comment bichonner son
            vélo"). Nous voulons aussi mettre en place une "stammtisch"
            (rencontre informelle) tous les mois à l’anneau
          </h4>
        </li>
        <li>
          <h4>
            Vous savez que sans Longchamp, la vie serait moins belle ! On a
            tendance à l’oublier car on l’a toujours connu mais avoir Longchamp
            est une chance extraordinaire. Il faut pouvoir 
            <b>protéger et pérenniser l’anneau</b>
          </h4>
        </li>
      </ol>
    </JoinContainer>
    <Faq>
      <h3>FAQ :</h3>
      <Question>
        Qui êtes-vous et pourquoi avoir créer une association ?
      </Question>
      <Answer>
        Nous sommes des cyclistes qui roulons régulièrement sur l’anneau de
        Longchamp. Nous avons voulu créer une association pour améliorer les
        conditions de sécurité et de confort de l’anneau
      </Answer>
      <Question>
        Je n’ai pas le temps de m’engager dans l’association mais je veux la
        soutenir, dois-je adhérer ?
      </Question>
      <Answer>
        Oui ! Une fois adhérent, chacun décide du temps qu’il peut et veut
        consacrer à l’association
      </Answer>
      <Question>
        Est-ce qu’il y a des conditions pour devenir adhérent ?
      </Question>
      <Answer>
        La seule condition est de partager notre vision qu’il faut défendre et
        promouvoir l’anneau de Longchamp pour que les cyclistes puissent
        pratiquer le sport qu’ils aiment dans les meilleures conditions
      </Answer>
      <Question>
        Je fais du vélotaf par l’anneau de Longchamp, puis-je adhérer ?
      </Question>
      <Answer>
        Bien sûr ! Nous sommes davantage orientés cyclisme sportif mais nous
        avons besoin que nos membres représentent toutes les formes de cyclisme,
        y compris les vélotafeurs qui sont une part non négligeable des usagers
        de l’anneau
      </Answer>
      <Question>Je viens très rarement à Longchamp, dois-je adhérer ?</Question>
      <Answer>
        Oui ! Votre adhésion aidera à améliorer les conditions de l’anneau pour
        les cyclistes réguliers ou plus occasionnels. Un occasionnel a souvent
        de meilleures idées qu’un cycliste qui connait chaque piège de Longchamp
        (revêtement abimé, journées plus compliquées)
      </Answer>
      <Question>
        Je suis déjà membre d’un autre club, puis-je adhérer ?
      </Question>
      <Answer>
        Oui et surtout pensez à faire adhérer votre Club (pour que nous
        puissions représenter encore plus de pratiquants)Oui et surtout pensez à
        faire adhérer votre Club (pour que nous puissions représenter encore
        plus de pratiquants)
      </Answer>
    </Faq>
  </>
)

export default Join
