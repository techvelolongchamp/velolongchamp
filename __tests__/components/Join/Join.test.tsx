import React from 'react'
import { RenderWithProvider as render } from '../../provider'

import Join from '../../../components/Join/Join'

describe('Join unit tests', () => {
  it('should render correctly', () => {
    const { container } = render(<Join />)

    expect(container).toMatchSnapshot()
  })
})
