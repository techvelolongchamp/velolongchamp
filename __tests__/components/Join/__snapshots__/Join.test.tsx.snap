// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Join unit tests should render correctly 1`] = `
.c1 {
  text-align: center;
  margin: 20px;
}

.c2 {
  background: #6d6e71;
  border-radius: 5px;
  color: #fff;
  cursor: pointer;
  font-size: 16px;
  margin: 20px 10px;
  padding: 10px;
}

.c0 {
  padding-bottom: 20px;
}

.c0 p {
  color: #0d171b;
  font-size: 14px;
  line-height: 20px;
  padding: 10px 0;
  text-align: justify;
}

.c0 ol {
  counter-reset: styled-counter;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  list-style: none;
  margin: 0;
  padding: 0;
}

.c0 ol li {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  counter-increment: styled-counter;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 10px 0;
  width: 100%;
}

.c0 ol li p {
  color: #6d6e71;
  font-size: 16px;
  font-weight: 500;
  line-height: 20px;
  padding-right: 10px;
}

.c0 ol li::before {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #6d6e71;
  content: '0' counter(styled-counter);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: 'Abril Fatface',serif;
  font-size: 30px;
  font-weight: bold;
  line-height: 1;
  margin-right: 10px;
}

.c3 {
  padding-bottom: 30px;
}

.c3 h3 {
  color: #9e9577;
  font-size: 16px;
  font-weight: 600;
  margin: 20px 0;
}

.c4 {
  color: #0d171b;
  font-size: 14px;
  font-style: italic;
  font-weight: 500;
  line-height: 20px;
  margin-top: 10px;
  padding: 10px 0;
}

.c5 {
  color: #0d171b;
  font-size: 14px;
  font-weight: 200;
  line-height: 20px;
  padding: 10px 0;
}

<div>
  <article
    class="c0"
  >
    <p>
      Devenir membre de VELO LONGCHAMP est gratuit et ne prend que quelques instants :
    </p>
    <div
      class="c1"
    >
      <a
        class="c2"
        href="/join"
      >
        Adhérer gratuitement
      </a>
    </div>
    <p>
      Voici 5 arguments imparables pour vous convaincre d’adhérer :
    </p>
    <ol>
      
    
      <li>
        <p>
          Vous roulerez plus tôt sur un 
          <strong>
            anneau sécurisé et confortable
          </strong>
           ! Car plus on est nombreux, plus on a de poids et plus vite seront faits les travaux d’aménagement nécessaires
        </p>
      </li>
      
    
      <li>
        <p>
          Votre 
          <strong>
            avis est tout aussi important
          </strong>
           que celui d’un autre ! En devenant membre, vous pouvez exprimer directement vos besoins et vos priorités ; vous pouvez lancer une initiative qui vous tient à cœur ; vous pouvez voter en Assemblée Générale ; vous êtes tenu au courant des avancées du projet
        </p>
      </li>
      
    
      <li>
        <p>
          Vous pouvez 
          <strong>
            cumuler avec d’autres clubs
          </strong>
           ! On n’est pas exclusif ! Vous pouvez rester membre de votre club de vélo, de triathlon ou autre.
        </p>
      </li>
      
    
      <li>
        <p>
          Vous ne roulerez 
          <strong>
            plus jamais seul
          </strong>
           ! En nous rejoignant, vous allez rencontrer d’autres cyclistes qui pourront, si vous le voulez, vous accompagner à Longchamp. Nous allons créer des ateliers (initiation tandem, "premiers secours" ou "comment bichonner son vélo"). Nous voulons aussi mettre en place une rencontre informelle tous les mois à l’anneau
        </p>
      </li>
      
    
      <li>
        <p>
          Vous savez que sans Longchamp, la vie serait moins belle ! On a tendance à l’oublier car on l’a toujours connu mais avoir Longchamp est une chance extraordinaire. Il faut pouvoir 
          <strong>
            protéger et pérenniser l’anneau
          </strong>
        </p>
      </li>
      
    
    </ol>
  </article>
  <article
    class="c3"
  >
    <h3>
      FAQ :
    </h3>
    
    
    <p
      class="c4"
    >
      Qui êtes-vous et pourquoi avoir créer une association ?
    </p>
    
    
    <p
      class="c5"
    >
      Nous sommes des cyclistes qui roulons régulièrement sur l’anneau de Longchamp. Nous avons voulu créer une association pour améliorer les conditions de sécurité et de confort de l’anneau
    </p>
    
    
    <p
      class="c4"
    >
      Je n’ai pas le temps de m’engager dans l’association mais je veux la soutenir, dois-je adhérer ?
    </p>
    
    
    <p
      class="c5"
    >
      Oui ! Une fois adhérent, chacun décide du temps qu’il peut et veut consacrer à l’association
    </p>
    
    
    <p
      class="c4"
    >
      Est-ce qu’il y a des conditions pour devenir adhérent ?
    </p>
    
    
    <p
      class="c5"
    >
      La seule condition est de partager notre vision qu’il faut défendre et promouvoir l’anneau de Longchamp pour que les cyclistes puissent pratiquer le sport qu’ils aiment dans les meilleures conditions
    </p>
    
    
    <p
      class="c4"
    >
      Je fais du vélotaf par l’anneau de Longchamp, puis-je adhérer ?
    </p>
    
    
    <p
      class="c5"
    >
      Bien sûr ! Nous sommes davantage orientés cyclisme sportif mais nous avons besoin que nos membres représentent toutes les formes de cyclisme, y compris les vélotafeurs qui sont une part non négligeable des usagers de l’anneau
    </p>
    
    
    <p
      class="c4"
    >
      Je viens très rarement à Longchamp, dois-je adhérer ?
    </p>
    
    
    <p
      class="c5"
    >
      Oui ! Votre adhésion aidera à améliorer les conditions de l’anneau pour les cyclistes réguliers ou plus occasionnels. Un occasionnel a souvent de meilleures idées qu’un cycliste qui connait chaque piège de Longchamp (revêtement abimé, journées plus compliquées)
    </p>
    
    
    <p
      class="c4"
    >
      Je suis déjà membre d’un autre club, puis-je adhérer ?
    </p>
    
    
    <p
      class="c5"
    >
      Oui et surtout pensez à faire adhérer votre Club (pour que nous puissions représenter encore plus de pratiquants)
    </p>
    
    
  </article>
</div>
`;
